packages/js/src/analysis/plugins/shortcode-plugin.js

```diff
@@ -0,0 +1,324 @@
+	 * @param {function} interface.registerPlugin       Register a plugin with Yoast SEO.
+	 * @param {function} interface.registerModification Register a modification with Yoast SEO.
+	 * @param {function} interface.pluginReady          Notify Yoast SEO that the plugin is ready.
+	 * @param {function} interface.pluginReloaded       Notify Yoast SEO that the plugin has been reloaded.
+	 * @param {Array}	shortcodesToBeParsed The array of shortcodes to be parsed.
+	 * @returns {void}
+	 */
+	constructor( { registerPlugin, registerModification, pluginReady, pluginReloaded }, shortcodesToBeParsed ) {
+		this._registerModification = registerModification;
+		this._pluginReady = pluginReady;
+		this._pluginReloaded = pluginReloaded;
+
+		registerPlugin( "YoastShortcodePlugin", { status: "loading" } );
+		this.bindElementEvents();
+
+		const keywordRegexString = "(" + shortcodesToBeParsed.join( "|" ) + ")";
+
+		// The regex for matching shortcodes based on the available shortcode keywords.
+		this.keywordRegex = new RegExp( keywordRegexString, "g" );
+		this.closingTagRegex = new RegExp( "\\[\\/" + keywordRegexString + "\\]", "g" );
+		this.nonCaptureRegex = new RegExp( "\\[" + keywordRegexString + "[^\\]]*?\\]", "g" );
+
+		/**
+		 * The array of parsedShortcode objects.
+		 *
+		 * @type {Object[]}
+		 */
+		this.parsedShortcodes = [];
+
+		this.loadShortcodes( this.declareReady.bind( this ) );
+	}
+
+	/* YOAST SEO CLIENT */
+
+	/**
+	 * Declares ready with YoastSEO.
+	 *
+	 * @returns {void}
+	 */
+	declareReady() {
+		this._pluginReady( "YoastShortcodePlugin" );
+		this.registerModifications();
+	}
+
+	/**
+	 * Declares reloaded with YoastSEO.
+	 *
+	 * @returns {void}
+	 */
+	declareReloaded() {
+		this._pluginReloaded( "YoastShortcodePlugin" );
+	}
+
+	/**
+	 * Registers the modifications for the content in which we want to replace shortcodes.
+	 *
+	 * @returns {void}
+	 */
+	registerModifications() {
+		this._registerModification( "content", this.replaceShortcodes.bind( this ), "YoastShortcodePlugin" );
+	}
+
+
+	/**
+	 * Removes all unknown shortcodes. Not all plugins properly registered their shortcodes in the WordPress backend.
+	 * Since we cannot use the data from these shortcodes they must be removed.
+	 *
+	 * @param {String} data The text to remove unknown shortcodes.
+	 * @returns {String} The text with removed unknown shortcodes.
+	 */
+	removeUnknownShortCodes( data ) {
+		data = data.replace( shortcodeStartRegex, "" );
+		data = data.replace( shortcodeEndRegex, "" );
+
+		return data;
+	}
+
+	/**
+	 * Replaces the unparsed shortcodes with the parsed ones.
+	 * The callback used to replace the shortcodes.
+	 *
+	 * @param {String} data The text to replace the shortcodes in.
+	 *
+	 * @returns {String} The text with replaced shortcodes.
+	 */
+	replaceShortcodes( data ) {
+		const parsedShortcodes = this.parsedShortcodes;
+
+		if ( typeof data === "string" && parsedShortcodes.length > 0 ) {
```

Comment:
The check on `parsedShortcodes.length > 0` is unnecessary. Also I don't feel it's necessary to do `const parsedShortcodes = this.parsedShortcodes`, right?